### ISSUE TITLE
Interactive TV buttons do not work

### ISSUE THREAD

**Issue by bennettpeter** [pot8oe/mythmote#89]
There are 4 buttons with red, green, yellow and blue dots. Pressing them has no effect. I tested and found that no message is sent to the MythTV frontend when they are pressed. Also holding the button down to assign it to a command does not work for these buttons. In this country (USA) we do not use the interactive buttons, so I do not understand what they are supposed to do. However, since they are there it would be nice if I could reassign them to some commands to be sent to MythTV.

**Comment by pot8oe** [comment]
Hi Peter,

In order to set the buttons' command you need to first go to settings and change the "Button Long-Press action" preference to "Edit Key-binding". You can then go back to the interactive TV buttons, long-press, and define the command that will be sent to the mythtv frontend.

The default commands should be:
Red - key f1
Green - key f2
Yellow - key f3
Blue - key f4

**Comment by bennettpeter** [comment]
Thank you for the prompt reply.

This is strange. I tried those things on a Samsung tablet. I could change key bindings for all buttons but when I held down the colored dot buttons, nothing happened. Then I tested by running this command on a computer
nc -k -l 6546
When I connect with mythmote to a computer running that, I can see all the commands sent by mythmote. With the Samsung tablet, nothing was displayed when I pressed the colored dots. Pressing other buttons showed the command on the computer.

Now I have installed mythmote on an android phone, and all is working as described. I get the dialog for changing the buttons when I hold them down, and the test with nc shows it is sending key f2, key f3, key f4, key f5 for the colored dot buttons.

Things seem to work differently on a phone or a tablet. On the tablet I went to play store and it said I had the latest version, and on the phone I just installed it today, so it is latest. On the tablet all buttons are shown at once on the screen, while on the phone they are on different pages. Maybe this has something to do with the problem on the tablet.

Also I notice that the phone keeps sending "query location", but the tablet did not. 

Anyway, it is fine for me, the phone is a better device to use for a remote than a tablet. If you want me to try anything more with the tablet to find out why that does not work with the colored dots, please let me know.

**Comment by pot8oe** [comment]
What version of Android is the Samsung tablet running?

You may try uninstalling and reinstalling mythmote. I wonder if the sqlite
db containing the button commands is corrupted.

I'll check to see if the tablet layout is broken in some way.

Thank you for the debugging info.

Best,
Tom

**Comment by bennettpeter** [comment]
The Samsung tablet is running Android 4.2.2, Kernel 3.0.31-1978026

I went into the application settings on the tablet and deleted all data for the application. Now the colored buttons are working correctly and the assignment dialog is working. 

One possible cause for the corruption - when I first installed Mythmote those colored buttons were not there. It must have done an update because suddenly one day they appeared, but were not working. I suspect that the upgrade process is not working correctly.

Regards
Peter
